### PR TITLE
Update Verification Method Revocation section.

### PR DESCRIPTION
**Pull request by msporny**
Partial editorial cleanup to appendices tracked as issue #728.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/742.html" title="Last updated on May 29, 2021, 7:33 PM UTC (365e0b3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/742/ab18a18...365e0b3.html" title="Last updated on May 29, 2021, 7:33 PM UTC (365e0b3)">Diff</a>

### CODE REVIEW

File: index.html
Comment (rhiaro) [inline index.html:4746]:
```suggestion
href="#method-operations">deactivate</a>, which might be used to remove
```

File: index.html
Comment (rhiaro) [inline index.html:4747]:
```suggestion
a <a>verification method</a> from a <a>DID document</a>.
```

File: index.html
Comment (rhiaro) [inline index.html:4774]:
```suggestion
revoked <a>verification method</a> is trickier than it might seem. Some <a>DID
```

**Review by rhiaro — CHANGES_REQUESTED** [review]

**Review by OR13 — APPROVED** [review]

**Review by brentzundel — APPROVED** [review]

File: index.html
Comment (dhh1128) [inline index.html:4719]: I believe I understand the intent of this sentence, and agree 100% with it. However, I think this sentence is very hard to parse, because of ambiguity in the phrase "applies only to the latest version." I suggest this be reworded to something along the lines of "Verification method revocation can only be embodied in changes to the latest version of a DID document; it cannot retroactively adjust previous versions."

File: index.html
Comment (dhh1128) [inline index.html:4736]: I disagree with this summary of the semantics of revocation. Previous text in this same section says that revocation can take place immediately after a key rotation to prevent the rotated key from a future compromise. This is at odds with the sentence here, which claims that revocation is understood as communicating doubt about the current security of the method.

Revocation does not always imply a belief that existing proofs or signatures might have been created by an attacker. Rather, it always implies an intention that future proofs or signatures shall not be creatable by an attacker, and it *may or may not* also imply a concern that existing proofs or signatures are suspect.

File: index.html
Comment (dhh1128) [inline index.html:4751]: This is at odds with the sentence that says that all DID methods support removing a method from a DID document as a form of revocation.

File: index.html
Comment (dhh1128) [inline index.html:4797]: I disagree with the phrase "only safe interpretation" -- because it implies that retroactive cancellation is safe. It isn't. Just like unwise use of revocation when the two constraining safety conditions (knowing when the cryptographic statement was made, and knowing what the state of the DID doc at that instant were), interpreting revocation to apply backward in time ALSO REWRITES HISTORY; the difference is that it rewrites that history to maximize protections for the controller instead of rewriting it to maximize protections for the recipients of cryptographically signed assertions. Either way, there's still a winner and a loser -- a protected party and a party treated cavalierly.

I do agree (strongly) with the third sentence, "DID ecosystems that take this approach..." What I would suggest as a revision would be to say, "When this is true, or when the state of a DID at the time of a cryptographically verifiable statement cannot be reliably determined, then the only safe course is to disallow any consideration DID state with respect to time, except the present moment. DID ecosystems that take this approach..."

What I am trying to do with this revision is to remove the phrase "apply both forward and backward in time" in its link to safety -- because I think describing it that way is problematic.

File: index.html
Comment (dhh1128) [inline index.html:4821]: I agree with this content in its narrow sense, but I don't like how it gives the impression (by focusing only on trustless systems) that the only way to achieve trust is to provide all these properties as part of the cryptographic inputs. The date that a mortgage was signed may not be part of the cryptographic inputs to a verifiable statement, and yet it may be possible to achieve full trust, by deciding that metadata about the event from some other source (e.g., banking records, independent witnesses) is also reliable. In fact, this is likely to happen in a courtroom in many scenarios: metadata from many sources will be evaluated and accepted/rejected. 

My suggestion is to add an extra paragraph right after the &lt;/ul&gt; that says something like: "In systems that are willing to admit metadata other than those constituting cryptographic input, similar trust may be achieved -- but always on the same basis (making a careful judgment about whether a DID document's content at the moment of a signing event authorized the verification method)."

**Review by dhh1128 — CHANGES_REQUESTED** [review]
See my specific comments inline.

**Review by dlongley — APPROVED** [review]
Approving provided we can address @dhh1128's concerns with some minor tweaks.

File: index.html
Comment (dlongley) [inline index.html:4739]:
```suggestion
Revocation is expected to be understood as a <a>controller</a> expressing that
proofs or signatures associated with a revoked <a>verification method</a>
created after its revocation should be treated as invalid. It could also imply
a concern that existing proofs or signatures may have been created by an
attacker, but this is not necessarily the case. Verifiers, however, might still
choose to accept or reject any such proofs or signatures at their own discretion.
```

File: index.html
Comment (dlongley) [inline index.html:4725]:
```suggestion
method is the only form of revocation that applies to all DID Methods that
support revocation.
```

File: index.html
Comment (dlongley) [reply index.html:4751]: I suggested a few words to remedy this above.

File: index.html
Comment (dlongley) [inline index.html:4800]: Resubmitted @dhh1128's text as a suggestion for ease of merging:

```suggestion
<a>DID</a>. When this is true, or when the state of a <a>DID</a> at the time of
a cryptographically verifiable statement cannot be reliably determined, then the
only safe course is to disallow any consideration of DID state with respect to
time, except the present moment. <a>DID</a> ecosystems that take this approach
essentially provide cryptographically verifiable statements as ephemeral tokens
that can be invalidated at any time by the <a>DID controller</a>.
```

File: index.html
Comment (dhh1128) [reply index.html:4751]: I like the suggestion.

File: index.html
Comment (dhh1128) [reply index.html:4739]: +1 for @dlongley 's suggestion.

File: index.html
Comment (dhh1128) [reply index.html:4725]: +1 for @dlongley 's suggestion.

File: index.html
Comment (kdenhartog) [inline index.html:4715]:
```suggestion
method</a> that is known to be compromised.
```

This change aligns with best practices described in [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280#section-5.3.1) which allow for the usage of a certificate hold status while investigating a compromise. This is quite useful in combination with certificate transparency lists.

**Review by kdenhartog — CHANGES_REQUESTED** [review]
Only one small nitpick from me. Will keep my eye on this PR to approve once it's merged/discuss with author.

File: index.html
Comment (msporny) [inline index.html:4720]:
```suggestion
<a>Verification method</a> revocation can only be embodied in changes to 
the latest version of a <a>DID Document</a>; it cannot retroactively adjust 
previous versions.
```

File: index.html
Comment (msporny) [inline index.html:4834]:
```suggestion
        </ul>
        <p>
In systems that are willing to admit metadata other than those constituting 
cryptographic input, similar trust may be achieved -- but always on the 
same basis where a careful judgment is made about whether a 
<a>DID document</a>'s content at the moment of a signing event 
contained the expected content.
        </p>
      </section>
```

File: index.html
Comment (msporny) [reply index.html:4738]: This was dealt with by a set of changes that @dlongley put forward, which were then agreed to by @dhh1128 and then merged.

File: index.html
Comment (msporny) [reply index.html:4825]: I added a minor variation of your language to the specification.

**Comment by msporny** [comment]
Editorial, multiple reviews, changes requested and made, no objections, merging.

**Comment by msporny** [comment]
@dhh1128, I'm fairly certain that we addressed all of your concerns in this PR so I merged it. If we didn't, please don't hesitate to request more changes (via another PR).

**Pull request merged**
